yt_dlp/extractor/gopro.py

```diff
@@ -0,0 +1,110 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from ..utils import (
+    int_or_none,
+    remove_end,
+    str_or_none,
+    try_get,
+    unified_timestamp,
+    url_or_none,
+)
+
+
+class GoProIE(InfoExtractor):
+    _VALID_URL = r'https?://(www\.)?gopro\.com/v/(?P<id>[A-Za-z0-9]+)'
+
+    _TESTS = [{
+        'url': 'https://gopro.com/v/ZNVvED8QDzR5V',
+        'info_dict': {
+            'id': 'ZNVvED8QDzR5V',
+            'title': 'My GoPro Adventure - 9/19/21',
+            'thumbnail': r're:https?://.+',
+            'ext': 'mp4',
+            'timestamp': 1632072947,
+            'upload_date': '20210919',
+            'uploader_id': 'fireydive30018',
+            'duration': 396062,
+        }
+    }, {
+        'url': 'https://gopro.com/v/KRm6Vgp2peg4e',
+        'info_dict': {
+            'id': 'KRm6Vgp2peg4e',
+            'title': 'じゃがいも カリカリ オーブン焼き',
+            'thumbnail': r're:https?://.+',
+            'ext': 'mp4',
+            'timestamp': 1607231125,
+            'upload_date': '20201206',
+            'uploader_id': 'dc9bcb8b-47d2-47c6-afbc-4c48f9a3769e',
+            'duration': 45187,
+            'track': 'The Sky Machine',
+        }
+    }, {
+        'url': 'https://gopro.com/v/kVrK9wlJvBMwn',
+        'info_dict': {
+            'id': 'kVrK9wlJvBMwn',
+            'title': 'DARKNESS',
+            'thumbnail': r're:https?://.+',
+            'ext': 'mp4',
+            'timestamp': 1594183735,
+            'upload_date': '20200708',
+            'uploader_id': '闇夜乃皇帝',
+            'duration': 313075,
+            'track': 'Battery (Live)',
+            'artist': 'Metallica',
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        metadata = self._parse_json(
+            self._html_search_regex(r'window\.__reflectData\s*=\s*([^;]+)', webpage, 'metadata'), video_id)
+
+        video_info = try_get(metadata, lambda x: x['collectionMedia'][0]) or {}
+        media_data = self._download_json(
+            'https://api.gopro.com/media/%s/download' % video_info.get('id'), video_id)
```

Comment:
```suggestion
        video_info = metadata['collectionMedia'][0]
        media_data = self._download_json(
            'https://api.gopro.com/media/%s/download' % video_info['id'], video_id)
```
If you are trying to make this non-fatal (to work with --ignore-no-formats-error), you need to make sure the download_json is also non-fatal. Otherwise, no reason to use try_get/.get here